src/main/io/osd.c

```diff
@@ -61,6 +59,177 @@
 
 #include "common/printf.h"
 
+#include "build/debug.h"
+
+#ifdef USE_DPRINTF
+extern serialPort_t *debugSerialPort;
+#define dprintf(x) if (debugSerialPort) printf x
```

Comment:
There are variadic macros, allowing much cleaner implementation: 
https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html

Or you can define full function (using vprintf internally), it will be optimized away when disabled